File: tauri-plugin-statusbar/ios/Sources/StatusbarPlugin.swift
import SwiftRs
import Tauri
import UIKit
import WebKit

struct SetDarkArgs: Decodable {
  let dark: Bool
}

class StatusbarPlugin: Plugin {
  // Match the system status bar (and any native UI) to the app theme by forcing
  // the window's interface style. With the status bar's default style this gives
  // light (white) icons in dark, dark icons in light.
  @objc public func setDark(_ invoke: Invoke) throws {
    let args = try invoke.parseArgs(SetDarkArgs.self)
    // Resolve from inside the main-thread block, after the change is applied.
    DispatchQueue.main.async {
      let style: UIUserInterfaceStyle = args.dark ? .dark : .light
      for scene in UIApplication.shared.connectedScenes {
        guard let windowScene = scene as? UIWindowScene else { continue }
        for window in windowScene.windows {
          window.overrideUserInterfaceStyle = style
        }
      }
      invoke.resolve()
    }
  }
}

@_cdecl("init_plugin_statusbar")
func initPlugin() -> Plugin {
  return StatusbarPlugin()
}

File: tauri-plugin-statusbar/android/src/main/java/StatusbarPlugin.kt
package com.luminaapps.talea.statusbar

import android.app.Activity
import androidx.core.view.WindowCompat
import app.tauri.annotation.Command
import app.tauri.annotation.InvokeArg
import app.tauri.annotation.TauriPlugin
import app.tauri.plugin.Invoke
import app.tauri.plugin.Plugin

@InvokeArg
class SetDarkArgs {
    var dark: Boolean = false
}

@TauriPlugin
class StatusbarPlugin(private val activity: Activity) : Plugin(activity) {
    @Command
    fun setDark(invoke: Invoke) {
        val args = invoke.parseArgs(SetDarkArgs::class.java)
        // Resolve from inside the UI-thread block, after the change is applied
        // (runOnUiThread posts asynchronously).
        activity.runOnUiThread {
            val window = activity.window
            val controller = WindowCompat.getInsetsController(window, window.decorView)
            // Dark app theme → light (white) bar icons; light theme → dark icons.
            controller.isAppearanceLightStatusBars = !args.dark
            controller.isAppearanceLightNavigationBars = !args.dark
            invoke.resolve()
        }
    }
}
